test/ably/restauth_test.py

```diff
@@ -202,6 +202,7 @@ async def test_authorize_returns_a_token_details(self):
         assert isinstance(token, TokenDetails)
 
     @dont_vary_protocol
+    @pytest.mark.filterwarnings('ignore::RuntimeWarning')
```

Comment:
I just realised this one isn't an internal deprecation warning (sorry, i missed this one when you posted them on slack). Could you have a look into why this warning is being emitted?

Reply:
It's emitted while calling the mocked method that was patched earlier in the test:
```
        with mock.patch('ably.rest.auth.Auth.request_token', new_callable=AsyncMock) as request_mock:
````
The method call that was patched is located in auth.py:99:
```
self.__token_details = await self.request_token(token_params, **auth_options)
```
Due to the patch, `AsyncMock.__call__` gets invoked but apparently it isn't awaited properly, which is classified as a `RuntimeWarning` since it's an `async def` as shown here:
```
class AsyncMock(mock.MagicMock):
    async def __call__(self, *args, **kwargs):
        return super(AsyncMock, self).__call__(*args, **kwargs)
```

Reply:
OK thanks for looking into this. The method is awaited in auth.py so it seems a bit odd that we get this warning. Any ideas how we can fix it?

Reply:
Yes, it does seem to be awaited, but it does not seem to have affect on the `AsyncMock.__call__` invocation.
When I replace the existing `AsyncMock` import (`test.ably.utils`) with the one from `unittest.mock`, the warning seems to go away.

Reply:
Seems my recent commits fixed the `RuntimeWarning` we discussed here, but upgrading `mock` from `1.3.0` to `4.0.3` broke another test in all Python versions (`test_cached_fallback`)

Reply:
Fixed it, it was a call that needed to be awaited after upgrading `mock` to `4.0.x`, but it wasn't required in earlier versions (<=3.x)